gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/privacySettings/js/EditPrivacySettings.js

```diff
@@ -0,0 +1,241 @@
+            "onCreate.fetchAuthPrefs": "{that}.fetchAvailableAuthorizedPrefs",
+            "onCreate.fetchDecisionPrefs": "{that}.fetchDecisionPrefs",
+            "afterRender.setCancelButtonText": {
+                "this": "{that}.dom.cancel",
+                method: "text",
+                args: "{that}.options.strings.cancel"
+            },
+            "afterRender.setDoneButtonText": {
+                "this": "{that}.dom.done",
+                method: "text",
+                args: "{that}.options.strings.done"
+            },
+            "afterRender.openDialog": {
+                listener: "gpii.oauth2.editPrivacySettings.openDialog",
+                args: ["{that}"]
+            },
+            "afterRender.bindCancel": {
+                "this": "{that}.dom.cancel",
+                method: "click",
+                args: "{that}.closeDialog"
+            },
+            "afterRender.bindDone": {
+                "this": "{that}.dom.done",
+                method: "click",
+                args: "{that}.savePrefsAndExit"
+            }
+        },
+        invokers: {
+            closeDialog: {
+                "this": "{that}.dialog",
+                method: "dialog",
+                args: ["close"]
+            },
+            fetchDecisionPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.fetchDecisionPrefs.url", {
+                    authDecisionId: "{that}.model.clientData.authDecisionId"
+                }, {
+                    type: "{that}.options.requestInfos.fetchDecisionPrefs.type",
+                    dataType: "json",
+                    success: "{that}.setDecisionPrefs"
+                }]
+            },
+            saveDecisionPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.saveDecisionPrefs.url", {
+                    authDecisionId: "{that}.model.clientData.authDecisionId"
+                }, {
+                    type: "{that}.options.requestInfos.saveDecisionPrefs.type",
+                    dataType: "json",
+                    contentType: "application/json",
+                    data: {
+                        expander: {
+                            funcName: "JSON.stringify",
+                            args: [{
+                                expander: {
+                                    func: "{selectionTree}.toServerModel",
+                                    args: "{selectionTree}.model"
+                                }
+                            }]
+                        }
+                    }
+                }]
+            },
+            fetchAvailableAuthorizedPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.fetchAvailableAuthorizedPrefs.url", {
+                    clientID: "{that}.model.clientData.oauth2ClientId"
+                }, {
+                    //TODO: Handle errors
```

Comment:
I've added a default error handler to gpii.oauth2.ajax:

https://github.com/simonbates/universal/commit/768ae7ed9481e537232378afc0704b7e2f61cc93